### PR TITLE
Add initial handling of Date type as Document id

### LINKED ISSUE #371: (Date/Time support) allow use of Date/Time as Document Id (`_id`)
For full Date/Time support (#362) it needs to be possible to specify Date/Time value as document id (`_id`), and allow use in `$eq` (and "$in") queries.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Adds support for EJSON-encoded Date values as typed Document Ids (`_id`).

**Which issue(s) this PR fixes**:
Fixes #371

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/InsertIntegrationTest.java [hunk 1]
```diff
@@ -148,6 +148,63 @@ public void insertDocumentWithDate() {
...
+      String expected =
+          """
+            {
+              "_id": {"$date": 1672539900000},
+              "username": "doc_date_user4",
+              "status": false
+            }
+            """;
+
+      String query_json =
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/InsertIntegrationTest.java:187]: @maheshrajamani This fails, not finding the document. I haven't been able to figure out why (DBFilterBase handling seems fine wrt DocumentIds) so may need help here.

Comment (maheshrajamani) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/InsertIntegrationTest.java:187]: @tatu-at-datastax Had to fix in FilterClauseDeserializer to resolve "_id" with date as DocumentId. It was resolving to Date. I have fixed it and checked in to this branch. Let me know if it is fine.

**Review by maheshrajamani — APPROVED** [review]
LGTM., Can you add a test in FindIntegrationTest to read based on date type document Id?

**Comment by tatu-at-datastax** [comment]
> LGTM., Can you add a test in FindIntegrationTest to read based on date type document Id?

Yes, will do that.

**Comment by tatu-at-datastax** [comment]
Accidentally closed, reopening to add test.

**Comment by tatu-at-datastax** [comment]
> LGTM., Can you add a test in FindIntegrationTest to read based on date type document Id?

@maheshrajamani Added one new doc in `FindIntegrationTest`, matching test. Ready for re-review.

**Review by vkarpov15 — APPROVED** [review]
LGTM :+1:

**Pull request merged**